### PR TITLE
Testing variables to prevent errors with PHP7.4

### PR DESCRIPTION
**Pull request by arnaudpn**

### CODE REVIEW

**Comment by fisharebest** [comment]
webtrees 1.7 does not run on PHP 7.4

There are very many compatibility issues with PHP 7.4

It only runs on PHP 5.3 - 7.3.  See the [server requirements](https://webtrees.net/install/requirements/)

**Comment by arnaudpn** [comment]
It runs quite well with PHP7.4 on my trees so far.
I'll continue submit patches in my own repo then.

**Comment by fisharebest** [comment]
> I'll continue submit patches in my own repo then.

I am happy to accept PRs on 1.7.  But I need to test them.
Can you explain how to reproduce each error.

File: app/Filter.php [hunk 1]
```diff
@@ -262,7 +262,7 @@ private static function inputArray($source, $variable, $regexp = null, $default
                 )
             );
 
-            return $tmp[$variable] ?: array();
+            if (is_array($tmp)) return $tmp[$variable] ?: array();
```
Comment (fisharebest) [inline app/Filter.php:265]: Code style: always use `{}` with an `if` statement.

File: app/Individual.php [hunk 2]
```diff
@@ -1154,7 +1154,8 @@ protected function addName($type, $full, $gedcom)
         // Extract the structured name parts - use for "sortable" names and indexes
         ////////////////////////////////////////////////////////////////////////////
 
-        $sublevel = 1 + (int) $gedcom[0];
+        $sublevel = 1;
+        if ($gedcom) $sublevel += (int) $gedcom[0];
```
Comment (fisharebest) [inline app/Individual.php:1158]: Code style: always use `{}` with an `if` statement.

Strict types: use `if ($gedcom !== '')` or `if ($gedcom !== null)` etc.

**Comment by arnaudpn** [comment]
app/Filter.php patch solves a white page when trying to open the page admin_site_clean.php (Control panel > Website > Clean up data folder).

The associated PHP error log is :

> /var/www/webtrees/webtrees_1.7.19/app/Filter.php:266 Trying to access array offset on value of type null
> #0 /app/Filter.php:266 ErrorException("Trying to access array offset on value of type null")
> #1 /app/Filter.php:376 inputArray(0, 'to_delete', [NULL], [NULL])
> #2 /admin_site_clean.php:24 postArray('to_delete')

**Comment by arnaudpn** [comment]
app/Individual.php patch solves the following error found in the logs:

> /var/www/webtrees/webtrees_1.7.19/app/Individual.php:1157 Trying to access array offset on value of type null
> #0 /app/Individual.php:1157 ErrorException("Trying to access array offset on value of type null")
> #1 /app/GedcomRecord.php:636 addName('INDI', 'Détails privés', [NULL])
> #2 /app/Query/QueryName.php:516 getAllNames()
> #3 /indilist.php:261 individuals([Tree], 'SAINT-VAUDELIN', 'S', '', false, false)

I couldn't reproduce this error by myself.
This error is being triggered by indexing bots IP's.

**Pull request closed without merging**